src/analysisd/state.c

```diff
@@ -319,6 +397,160 @@ void w_get_initial_queues_size() {
+    w_mutex_lock(&agents_state_mutex);
+    analysisd_agent_state_t *agent_node = get_node(agent_id);
+    agent_node->events_decoded_breakdown.hostinfo++;
+    w_mutex_unlock(&agents_state_mutex);
+}
+
+static void w_inc_winevt_agent_decoded_events(char * agent_id) {
+    w_mutex_lock(&agents_state_mutex);
+    analysisd_agent_state_t *agent_node = get_node(agent_id);
+    agent_node->events_decoded_breakdown.winevt++;
+    w_mutex_unlock(&agents_state_mutex);
+}
+
+static void w_inc_dbsync_agent_decoded_events(char * agent_id) {
+    w_mutex_lock(&agents_state_mutex);
+    analysisd_agent_state_t *agent_node = get_node(agent_id);
+    agent_node->events_decoded_breakdown.dbsync++;
+    w_mutex_unlock(&agents_state_mutex);
+}
+
+static void w_inc_upgrade_agent_decoded_events(char * agent_id) {
+    w_mutex_lock(&agents_state_mutex);
+    analysisd_agent_state_t *agent_node = get_node(agent_id);
+    agent_node->events_decoded_breakdown.upgrade++;
+    w_mutex_unlock(&agents_state_mutex);
+}
+
+static void w_inc_events_agent_decoded(char * agent_id) {
+    w_mutex_lock(&agents_state_mutex);
+    analysisd_agent_state_t *agent_node = get_node(agent_id);
+    agent_node->events_decoded_breakdown.events++;
+    w_mutex_unlock(&agents_state_mutex);
+}
+
+static void w_inc_processed_agent_events(char * agent_id) {
+    w_mutex_lock(&agents_state_mutex);
+    analysisd_agent_state_t *agent_node = get_node(agent_id);
+    agent_node->events_processed++;
+    w_mutex_unlock(&agents_state_mutex);
+}
+
+static void w_inc_alerts_agent_written(char * agent_id) {
+    w_mutex_lock(&agents_state_mutex);
+    analysisd_agent_state_t *agent_node = get_node(agent_id);
+    agent_node->alerts_written++;
+    w_mutex_unlock(&agents_state_mutex);
+}
+
+static void w_inc_archives_agent_written(char * agent_id) {
+    w_mutex_lock(&agents_state_mutex);
+    analysisd_agent_state_t *agent_node = get_node(agent_id);
+    agent_node->archives_written++;
+    w_mutex_unlock(&agents_state_mutex);
+}
+
+static void w_inc_firewall_agent_written(char * agent_id) {
+    w_mutex_lock(&agents_state_mutex);
+    analysisd_agent_state_t *agent_node = get_node(agent_id);
+    agent_node->firewall_written++;
+    w_mutex_unlock(&agents_state_mutex);
+}
+
+static analysisd_agent_state_t * get_node(char *agent_id){
+    analysisd_agent_state_t * agent_state = (analysisd_agent_state_t *) OSHash_Get_ex(analysisd_agents_state, agent_id);
+
+    if(agent_state != NULL) {
+        return agent_state;
+    } else {
+        os_calloc(1,sizeof(analysisd_agent_state_t),agent_state);
+        agent_state->id = atoi(agent_id);
+        OSHash_Add_ex(analysisd_agents_state, agent_id, agent_state);
+        return agent_state;
+    }
+}
+
+static void w_analysisd_clean_agents_state() {
+    int *active_agents = NULL;
+    int sock = -1;
+    OSHashNode *hash_node;
+    unsigned int inode_it = 0;
+
+    w_mutex_lock(&agents_state_mutex);
```

Comment:
Can we first get the information from the DB and at the end lock and update the counters structure?